### ISSUE TITLE
/etc/ld.so.conf docker import

### ISSUE THREAD

**Issue by sachnk** [apptainer/singularity#560]
I'm creating a CentOS7 based singularity image from an existing docker image using:

`singularity import myimage.img docker://xyz/abc:latest`

The Dockerfile that created xyz/abc:latest modifies the /etc/ld.so.conf so that it contains /usr/local/lib and also runs ldconfig. If I run a shell in the docker image, I see:

`$ cat /etc/ld.so.conf`
`include ld.so.conf.d/*.conf`
`/usr/local/lib`

However, when I import the docker image into a new singularity image, the changes I made to /etc/ld.so.conf are not there:

`$ cat /etc/ld.so.conf`
`include ld.so.conf.d/*.conf`

Why aren't my changes propagated into singularity? Any ideas?

Thanks

**Comment by vsoch** [comment]
We need to try this when @gmkurtzer merges the PR #553, as there was an issue with Docker layers. Then we can further debug. Just to be clear, the docker:// endpoint pulls from Docker hub so making changes to a local image will not show up when you import!

**Comment by sachnk** [comment]
Understood, thanks. Point taken regarding the docker:// endpoint using
Docker Hub -- we publish to Docker Hub first before converting to
singularity.

I've worked around this by manually modifying the singularity image for
now, but definitely look forward to the fix so our conversions from Docker
are seamless.

Thanks!

**Comment by sachnk** [comment]
@vsoch @gmkurtzer Is there a time line when PR #553 might make it into master / release tag?

**Comment by vsoch** [comment]
It's in development now if you want to test out!

**Comment by vsoch** [comment]
you can do:

```bash
git clone https://github.com/singularityware/singularity
cd singularity
checkout -b development
git pull origin development
```

then install per usual

```bash
./autogen.sh
./configure --prefix=/use/local
make
sudo make install
```

**Comment by sachnk** [comment]
Hi @vsoch,

Yep, I pulled development and created an image with a new singularity binary. The new image seems to work correctly -- I see the right contents in /etc/ld.so.conf.

It also corrects some other issues I found after posting the ld.so.conf issue -- namely, missing packages from pip installs even though the Docker image has them.

Unless you advise otherwise, I'm going to go ahead and use this development version of singularity.

Thanks!
